apps/api/src/app/environments/usecases/validate-mx-record/validate-mx-record.usecase.ts

```diff
@@ -0,0 +1,58 @@
+import { Injectable, Scope } from '@nestjs/common';
+import { EnvironmentRepository } from '@novu/dal';
+import * as dns from 'dns';
+import { IJwtPayload } from '@novu/shared';
+import { EnvironmentEntity } from '@novu/dal';
+
+@Injectable({
+  scope: Scope.REQUEST,
+})
+export class ValidateMxRecord {
+  constructor(private environmentRepository: EnvironmentRepository) {}
+
+  async execute(command: IJwtPayload): Promise<any> {
+    const env = await this.environmentRepository.findOne({ _id: command.environmentId });
+
+    const domain = env.dns?.domain;
+
+    if (!domain) return false;
```

Comment:
So I am not sure about it because the request is good but the inboundParseDomain is missing, at the moment i kept the response with { mxRecordConfigured: false } 
We could add more information on the client side/ docs on the right way to configure it, wdyt?